library_generation/generate_composed_library.py

```diff
@@ -0,0 +1,139 @@
+  In this case, generate_library.sh will be called once with
+    generate_library.sh --proto_path google/cloud/asset/v1 --destination_path google-cloud-asset-v1-java
+    and once with
+    generate_library.sh --proto_path google/cloud/asset/v1p2bet5 --destination_path google-cloud-asset-v1p2beta5-java
+--versions_file: a file of newline-separated version strings in the form "module:released-version:current-version". The versions will be applied to the pom.xml files and readmes
+
+Prerequisites
+- Needs an `output` folder at the location of the calling shell. This folder
+is automatically detected by `generate_library.sh` and this script ensures it
+contains the necessary folders and files, specifically:
+  - A "google" folder found in the googleapis repository
+  - A "grafeas" folder found in the googleapis repository
+Note: googleapis repo is found in https://github.com/googleapis/googleapis.
+"""
+
+import click
+import utilities as util
+import os
+import subprocess
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+
+@click.group(invoke_without_command=False)
+@click.pass_context
+@click.version_option(message="%(version)s")
+def main(ctx):
+    pass
+
+@main.command()
+@click.option(
+    "--generation_queries",
+    required=True,
+    type=str,
+    help="""
+      a single string of comma-separated key-value groups separated by a
+      pipe | (i.e. the groups are spearated by pipe, while a group's key-values are
+      separated by comma). They key-value groups are in the form of `key=value` and will
+      be converted to an argument to generate_library.sh (`--key value`).
+
+      example: "proto_path=google/cloud/asset/v1,destination_path=google-cloud-asset-v1-java,(...)|proto_path=google/cloud/asset/v1p2beta5,destination_path=google-cloud-asset-v1-java,(...)"
+      In this case, generate_library.sh will be called once with
+        generate_library.sh --proto_path google/cloud/asset/v1 --destination_path google-cloud-asset-v1-java
+        and once with
+        generate_library.sh --proto_path google/cloud/asset/v1p2bet5 --destination_path google-cloud-asset-v1p2beta5-java
+    """
+)
+@click.option(
+    "--repository_path",
+    required=True,
+    type=str,
+    help="""
+    A relative path from the output folder (found in cwd when calling this script) that points
+    to the location of the original library. The original library is generally downloaded from
+    github (example, google-cloud-java/java-asset). Postprocessing will be run on such folder,
+    once the pre-processed libraries have been generated
+    """
+)
+@click.option(
+    "--versions_file",
+    required=False,
+    type=str,
+    help="""
+    a file of newline-separated version strings in the form "module:released-version:current-version". 
+    The versions will be applied to the pom.xml files and readmes
+    """
+)
+@click.option(
+    "--enable_postprocessing",
```

Comment:
Is there a case whether postprocessing is not needed? In OwlBot service, one of postprocessor's job is to bring files from the owl-bot-staging (So postprocessing is always required when copying generated code). What is "postprocessing" in this script now?

Reply:
For self-service in the future, we probably won't have post-processing enabled.